File: src/php/src/PaymentWS.php
<?php
/**
 * jbilling-php-api
 * Copyright (C) 2007-2009  Make A Byte, inc
 * http://www.makeabyte.com
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 * 
 * @package com.makeabyte.contrib.jbilling.php
 */

/**
  * PaymentWS
  * @author Jeremy Hahn
  * @version 1.0
  * @copyright Make A Byte, inc
  * @package com.makeabyte.contrib.jbilling.php
  */

class PaymentWS {

      var $id;              // Unique identifier of the paymentrecord
	  var $amount;          // The amount of the payment operation.
	  var $balance;         // Balance of this payment. If greater than 0, this payment could pay part of another invoice. If 0, this payment has already been applied to an invoice, lowering the invoice's balance.
	  var $createDateTime;  // Date in which this payment record was created.
	  var $updateDateTime;  // Date in which this payment record was last updated.
	  var $paymentDate;     // Date of the payment.
	  var $attempt;         // Number of the attempt to process this payment
	  var $deleted;         // “1” if this record has been deleted, “0” otherwise.
      var $methodId;        // Identifier of the payment method. Refer to Appendix A for a list of acceptable values.
      var $resultId;        // Identifier of the result of the payment attempt. Refer to Appendix A for a list of acceptable values.
      var $isRefund;        // Integer “1” if this payment constitutes a refund operation, “0” otherwise.
      var $isPreauth;       // “1” if this payment is a pre-authorization, “0” otherwise.
      var $currencyId;      // Identifier of the currency in which the payment is being made. See Appendix A for a list of acceptable values.
      var $userId;          // Identifier of the user this payment record belongs to.
      var $cheque;          // If this payment is done via check, this property contains information about the cheque, otherwise it contains "null”.
      var $creditCard;      // If this is a credit card payment, this property contains information about the credit card, otherwise it contains "null”.
      var $ach;             // If this is a payment done with Automatic Clearing House (ACH), this property contains the banking information needed.
      var $method;          // Name of the payment method used.
      var $invoiceIds;      // Contains the list of invoices this payment is paying.
      var $paymentId;       // Refund specific field. When a refund is to be issued, this field holds the identifier of the payment that is to be refunded.
      var $authorizationId;   // Refund specific field. Contains the DTO authorization field for the refund.

      /**
       * The PaymentWS constructor
       * 
       * @access public
       */
	  public function PaymentWS() {
	  }
	  /**
	   * Sets the id property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $id Unique identifier of the paymentrecord
	   * @return void
	   */
	  public function setId( $id ) {
	  	
	  	     $this->id = $id;
	  }
	  /**
	   * Sets the amount property of the PaymentWS object
	   * 
	   * @access public
	   * @param Float $amount The amount of the payment operation.
	   * @return void
	   */
	  public function setAmount( $amount ) {
	  	
	  	     $this->amount = $amount;
	  }
	  /**
	   * Sets the balance property of the PaymentWS object
	   * 
	   * @access public
	   * @param Float $bal Balance of this payment. If greater than 0, this payment could pay part of another invoice. If 0, this payment has already been applied to an invoice, lowering the invoice's balance.
	   * @return void
	   */
	  public function setBalance( $bal ) {
	  	
	  	     $this->balance = $bal;
	  }
	  /**
	   * Sets the createDateTime property of the PaymentWS object
	   * 
	   * @access public
	   * @param Date $date Date in which this payment record was created.
	   * @return void
	   */
	  public function setCreateDateTime( $date ) {
	  	
	  	     $this->createDateTime = $date;
	  }
	  /**
	   * Sets the updateDateTime property of the PaymentWS object
	   * 
	   * @access public
	   * @param Date $date Date in which this payment record was last updated.
	   * @return void
	   */
	  public function setUpdateDateTime( $date ) {
	  
	         $this->updateDateTime = $date;
	  }
	  /**
	   * Sets the paymentDate property of the PaymentWS object
	   * 
	   * @access public
	   * @param Date $date Date of the payment.
	   * @return void
	   */
	  public function setPaymentDate( $date ) {
	  	
	  	     $this->paymentDate = $date;
	  }
	  /**
	   * Sets the attempt property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $int Number of the attempt to process this payment
	   * @return void
	   */
	  public function setAttempt( $attempt ) {

	  	     $this->attempt = $attempt;
	  }
	  /**
	   * Sets the deleted property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $int “1” if this record has been deleted, “0” otherwise.
	   * @return void
	   */
	  public function setDeleted( $deleted ) {
	  	
	  	     $this->deleted = $deleted;
	  }
	  /**
	   * Sets the methodId property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $id Identifier of the payment method. Refer to Appendix A for a list of acceptable values.
	   * @return void
	   */
	  public function setMethodId( $id ) {
	  	
	  	     $this->methodId = $id;
	  }
	  /**
	   * Sets the resultId property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $id Identifier of the result of the payment attempt. Refer to Appendix A for a list of acceptable values.
	   * @return void
	   */
	  public function setResultId( $id ) {
	  	
	  	     $this->resultId = $id;
	  }
	  /**
	   * Sets the isRefund property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $id Integer “1” if this payment constitutes a refund operation, “0” otherwise.
	   * @return void
	   */
	  public function setIsRefund( $int ) {
	  	
	  	     $this->isRefund = $int;
	  }
	  /**
	   * Sets the isPreAuth property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $id “1” if this payment is a pre-authorization, “0” otherwise.
	   * @return void
	   */
	  public function setIsPreAuth( $int ) {
	  	
	  	     $this->isPreauth = $int;
	  }
	  /**
	   * Sets the currencyId property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $id Identifier of the currency in which the payment is being made. See Appendix A for a list of acceptable values. 
	   * @return void
	   */
	  public function setCurrencyId( $id ) {
	  	
	  	     $this->currencyId = $id;
	  }
	  /**
	   * Sets the userId property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $id Identifier of the user this payment record belongs to. 
	   * @return void
	   */
	  public function setUserId( $id ) {
	  	
	  	     $this->userId = $id;
	  }
	  /**
	   * Sets the cheque property of the PaymentWS object
	   * 
	   * @access public
	   * @param PaymentInfoChequeDTO $cheque If this payment is done via check, this property contains information about the cheque, otherwise it contains "null”. 
	   * @return void
	   */
	  public function setCheque( $cheque ) {
	  	
	  	     $this->cheque = $cheque;
	  }
	  /**
	   * Sets the creditCard property of the PaymentWS object
	   * 
	   * @access public
	   * @param CreditCardDTO $cc If this is a credit card payment, this property contains information about the credit card, otherwise it contains "null”. 
	   * @return void
	   */
	  public function setCreditCard( $cc ) {
	  	
	  	     $this->creditCard = $cc;
	  }
	  /**
	   * Sets the ach property of the PaymentWS object
	   * 
	   * @access public
	   * @param  AchDTO $ach If this is a payment done with Automatic Clearing House (ACH), this property contains the banking information needed.
	   * @return void
	   */
	  public function setAch( $ach ) {
	  	
	  	     $this->ach = $ach;
	  }
	  /**
	   * Sets the method property of the PaymentWS object
	   * 
	   * @access public
	   * @param String $method Name of the payment method used.
	   * @return void
	   */
	  public function setMethod( $method ) {

	  	     $this->method = $method;
	  }
	  /**
	   * Sets the invoice property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer[] $id Contains the list of invoices this payment is paying.
	   * @return void
	   */
	  public function setInvoiceId( $id ) {
	  	
	  	     $this->invoiceId = $id;
	  }
	  /**
	   * Sets the payment property of the PaymentWS object
	   * 
	   * @access public
	   * @param Integer $id Refund specific field. When a refund is to be issued, this field holds the identifier of the payment that is to be refunded.
	   * @return void
	   */
	  public function setPaymentId( $id ) {
	  	     
	  	     $this->paymentId = $id;
	  }
	  /**
	   * Sets the payment property of the PaymentWS object
	   * 
	   * @access public
	   * @param PaymentAuthorizationDTO $auth Refund specific field. Contains the DTO authorization field for the refund.
	   * @return void
	   */
	  public function setAuthorization( $auth ) {
	  	
	  	    $this->authorizationId = $auth;
	  } 
	  /**
	   * Gets the id property of the PaymentWS object
	   * 
	   * @access public
	   * @return Unique identifier of the paymentrecord
	   */
	  public function getId() {

	  	     return $this->id;
	  }
	  /**
	   * Gets the amount property of the PaymentWS object
	   * 
	   * @access public
	   * @return The amount of the payment operation.
	   */
	  public function getAmount() {

	  	     return $this->amount;
	  }
	  /**
	   * Gets the balance property of the PaymentWS object
	   * 
	   * @access public
	   * @return Balance of this payment. If greater than 0, this payment could pay part of another invoice. If 0, this payment has already been applied to an invoice, lowering the invoice's balance.
       */
	  public function getBalance() {

	  	     return $this->balance;
	  }
	  /**
	   * Gets the createDateTime property of the PaymentWS object
	   * 
	   * @access public
	   * @return Date in which this payment record was created.
	   */
	  public function getCreateDateTime() {

	  	     return $this->createDateTime;
	  }
	  /**
	   * Gets the updateDateTime property of the PaymentWS object
	   * 
	   * @access public
	   * @return Date in which this payment record was last updated.
       */
	  public function getUpdateDateTime() {

	         return $this->updateDateTime;
	  }
	  /**
	   * Gets the paymentDate property of the PaymentWS object
	   * 
	   * @access public
	   * @return Date of the payment.
	   */
	  public function getPaymentDate() {

	  	     return $this->paymentDate;
	  }
	  /**
	   * Gets the attempt property of the PaymentWS object
	   * 
	   * @access public
	   * @return Number of the attempt to process this payment
	   */
	  public function getAttempt() {

	  	     return $this->attempt;
	  }
	  /**
	   * Gets the deleted property of the PaymentWS object
	   * 
	   * @access public
	   * @return “1” if this record has been deleted, “0” otherwise.
	   */
	  public function getDeleted() {

	  	     return $this->deleted;
	  }
	  /**
	   * Gets the methodId property of the PaymentWS object
	   * 
	   * @access public
	   * @return Identifier of the payment method. Refer to Appendix A for a list of acceptable values.
	   */
	  public function getMethodId() {

	  	     return $this->methodId;
	  }
	  /**
	   * Gets the resultId property of the PaymentWS object
	   * 
	   * @access public
	   * @return Identifier of the result of the payment attempt. Refer to Appendix A for a list of acceptable values.
	   */
	  public function getResultId() {

	  	     return $this->resultId;
	  }
	  /**
	   * Gets the isRefund property of the PaymentWS object
	   * 
	   * @access public
	   * @return Integer “1” if this payment constitutes a refund operation, “0” otherwise.
	   */
	  public function getIsRefund() {

	  	     return $this->isRefund;
	  }
	  /**
	   * Gets the isPreAuth property of the PaymentWS object
	   * 
	   * @access public
	   * @return “1” if this payment is a pre-authorization, “0” otherwise.
	   */
	  public function getIsPreAuth() {

	  	     return $this->isPreauth;
	  }
	  /**
	   * Gets the currencyId property of the PaymentWS object
	   * 
	   * @access public
	   * @return Identifier of the currency in which the payment is being made. See Appendix A for a list of acceptable values. 
	   */
	  public function getCurrencyId() {

	  	     return $this->currencyId;
	  }
	  /**
	   * Gets the userId property of the PaymentWS object
	   * 
	   * @access public
	   * @return Identifier of the user this payment record belongs to. 
	   */
	  public function getUserId() {

	  	     return $this->userId;
	  }
	  /**
	   * Gets the cheque property of the PaymentWS object
	   * 
	   * @access public
	   * @return If this payment is done via check, this property contains information about the cheque, otherwise it contains "null”. 
	   */
	  public function getCheque() {

	  	     return $this->cheque;
	  }
	  /**
	   * Gets the creditCard property of the PaymentWS object
	   * 
	   * @access public
	   * @return If this is a credit card payment, this property contains information about the credit card, otherwise it contains "null”. 
	   */
	  public function getCreditCard() {

	  	     return $this->creditCard;
	  }
	  /**
	   * Gets the ach property of the PaymentWS object
	   * 
	   * @access public
	   * @return If this is a payment done with Automatic Clearing House (ACH), this property contains the banking information needed.
	   */
	  public function getAch() {

	  	     return $this->ach;
	  }
	  /**
	   * Gets the method property of the PaymentWS object
	   * 
	   * @access public
	   * @return Name of the payment method used.
	   */
	  public function getMethod() {

	  	     return $this->method;
	  }
	  /**
	   * Gets the invoice property of the PaymentWS object
	   * 
	   * @access public
	   * @return Contains the list of invoices this payment is paying.
	   */
	  public function getInvoiceId() {

	  	     return $this->invoiceId;
	  }
	  /**
	   * Gets the payment property of the PaymentWS object
	   * 
	   * @access public
	   * @return Refund specific field. When a refund is to be issued, this field holds the identifier of the payment that is to be refunded.
	   */
	  public function getPaymentId() {

	  	     return $this->paymentId;
	  }
	  /**
	   * Gets the payment property of the PaymentWS object
	   * 
	   * @access public
	   * @return Refund specific field. Contains the DTO authorization field for the refund.
	   */
	  public function getAuthorization() {

	  	    return $this->authorizationId;
	  }
}
?>

File: src/java/com/sapienter/jbilling/server/payment/PaymentWS.java
/*
 * JBILLING CONFIDENTIAL
 * _____________________
 *
 * [2003] - [2012] Enterprise jBilling Software Ltd.
 * All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Enterprise jBilling Software.
 * The intellectual and technical concepts contained
 * herein are proprietary to Enterprise jBilling Software
 * and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden.
 */

package com.sapienter.jbilling.server.payment;

import com.sapienter.jbilling.server.entity.AchDTO;
import com.sapienter.jbilling.server.entity.CreditCardDTO;
import com.sapienter.jbilling.server.entity.PaymentAuthorizationDTO;
import com.sapienter.jbilling.server.entity.PaymentInfoChequeDTO;
import com.sapienter.jbilling.server.metafields.MetaFieldValueWS;
import com.sapienter.jbilling.server.security.WSSecured;
import org.hibernate.validator.constraints.NotEmpty;

import javax.validation.Valid;
import javax.validation.constraints.Digits;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;


/**
 * @author Emil
 */
public class PaymentWS implements WSSecured, Serializable {

    @NotNull(message="validation.error.notnull")
    private Integer userId = null;

    @Valid
    private PaymentInfoChequeDTO cheque = null;
    @Valid
    private CreditCardDTO creditCard = null;
    @Valid
    private AchDTO ach = null;

    private String method = null;
    private Integer invoiceIds[] = null;

    // refund specific fields
    private Integer paymentId = null; // this is the payment refunded / to refund
    private PaymentAuthorizationDTO authorization = null;

    //missing properties from PaymentDTO
    @NotEmpty(message="validation.error.notnull")
    @Digits(integer=12, fraction=10, message="validation.error.not.a.number")
    private String amount;
    @NotNull(message="validation.error.notnull")
    private Integer isRefund;
    @NotNull(message="validation.error.apply.without.method")
    private Integer paymentMethodId;
    @NotNull(message="validation.error.notnull")
    private Date paymentDate;
    @NotNull(message="validation.error.notnull")
    private Integer currencyId;
    private int id;
    private Integer isPreauth;
    private Integer attempt;
    private String balance;
    private Date createDatetime;
    private Date updateDatetime;
    private int deleted;
    private Integer resultId;
    @Size(min = 0, max = 500, message = "validation.error.size,0,500")
    private String paymentNotes = null;
    private Integer paymentPeriod;

    @Valid
    private MetaFieldValueWS[] metaFields;
    
    public Integer getResultId() {
        return resultId;
    }

    public void setResultId(Integer resultId) {
        this.resultId = resultId;
    }
    
    public PaymentWS() {
        super();
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public PaymentInfoChequeDTO getCheque() {
        return cheque;
    }

    public void setCheque(PaymentInfoChequeDTO cheque) {
        this.cheque = cheque;
    }

    public CreditCardDTO getCreditCard() {
        return creditCard;
    }

    public void setCreditCard(CreditCardDTO creditCard) {
        this.creditCard = creditCard;
    }

    public AchDTO getAch() {
        return ach;
    }

    public void setAch(AchDTO ach) {
        this.ach = ach;
    }

    public String getMethod() {
        return method;
    }

    public void setMethod(String method) {
        this.method = method;
    }

    public Integer[] getInvoiceIds() {
        return invoiceIds;
    }

    public void setInvoiceIds(Integer[] invoiceIds) {
        this.invoiceIds = invoiceIds;
    }

    public Integer getPaymentId() {
        return paymentId;
    }

    public void setPaymentId(Integer paymentId) {
        this.paymentId = paymentId;
    }

    public PaymentAuthorizationDTO getAuthorizationId() {
        return authorization;
    }

    public void setAuthorization(PaymentAuthorizationDTO authorization) {
        this.authorization = authorization;
    }

    // required by CXF
    public void setAuthorizationId(PaymentAuthorizationDTO authorization) {
        this.authorization = authorization;
    }

    public String getAmount() {
        return amount;
    }

    public BigDecimal getAmountAsDecimal() {
        return amount != null ? new BigDecimal(amount) : null;
    }

    public void setAmountAsDecimal(BigDecimal amount) {
        setAmount(amount);
    }

    public void setAmount(String amount) {
        this.amount = amount;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = (amount != null ? amount.toString() : null);
    }

    public Integer getIsRefund() {
        return isRefund;
    }

    public void setIsRefund(Integer isRefund) {
        this.isRefund = isRefund;
    }

    public Integer getMethodId() {
        return paymentMethodId;
    }

    public void setMethodId(Integer paymentMethodId) {
        this.paymentMethodId = paymentMethodId;
    }

    public Date getPaymentDate() {
        return paymentDate;
    }

    public void setPaymentDate(Date paymentDate) {
        this.paymentDate = paymentDate;
    }

    public Integer getCurrencyId() {
        return currencyId;
    }

    public void setCurrencyId(Integer currencyId) {
        this.currencyId = currencyId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Integer getIsPreauth() {
        return isPreauth;
    }

    public void setIsPreauth(Integer isPreauth) {
        this.isPreauth = isPreauth;
    }

    public Integer getAttempt() {
        return attempt;
    }

    public void setAttempt(Integer attempt) {
        this.attempt = attempt;
    }

    public String getBalance() {
        return balance;
    }

    public BigDecimal getBalanceAsDecimal() {
        return balance != null ? new BigDecimal(balance) : null;
    }

    public void setBalanceAsDecimal(BigDecimal balance) {
        setBalance(balance);
    }

    public void setBalance(String balance) {
        this.balance = balance;
    }

    public void setBalance(BigDecimal balance) {
        this.balance = (balance != null ? balance.toString() : null);        
    }

    public Date getCreateDatetime() {
        return createDatetime;
    }

    public void setCreateDatetime(Date createDatetime) {
        this.createDatetime = createDatetime;
    }

    public Date getUpdateDatetime() {
        return updateDatetime;
    }

    public void setUpdateDatetime(Date updateDatetime) {
        this.updateDatetime = updateDatetime;
    }

    public int getDeleted() {
        return deleted;
    }

    public void setDeleted(int deleted) {
        this.deleted = deleted;
    }

    public void setPaymentNotes(String paymentNotes){
        this.paymentNotes = paymentNotes;
    }
    
    public String getPaymentNotes(){
        return paymentNotes;
    }
    
    public void setPaymentPeriod(Integer paymentPeriod){
        this.paymentPeriod = paymentPeriod;
    }
    
    public Integer getPaymentPeriod(){
        return paymentPeriod;
    }

    public MetaFieldValueWS[] getMetaFields() {
        return metaFields;
    }

    public void setMetaFields(MetaFieldValueWS[] metaFields) {
        this.metaFields = metaFields;
    }

    /**
     * Unsupported, web-service security enforced using {@link #getOwningUserId()}
     * @return null
     */
    public Integer getOwningEntityId() {
        return null;
    }

    public Integer getOwningUserId() {
        return getUserId();
    }

    @Override
    public String toString() {
        return "PaymentWS{"
               + "id=" + id
               + ", userId=" + userId
               + ", paymentMethodId=" + paymentMethodId
               + ", method='" + method + '\''
               + ", amount='" + amount + '\''
               + ", balance='" + balance + '\''
               + ", isRefund=" + isRefund
               + ", isPreauth=" + isPreauth
               + ", paymentDate=" + paymentDate
               + ", deleted=" + deleted
               + ", paymentId=" + paymentId
               + ", currencyId=" + currencyId
               + '}';
    }
}
